### ISSUE TITLE
Retroachievements on RetroArch for 3DS causes massive stutters

### ISSUE THREAD

**Issue by shigutso** [libretro/RetroArch#17027]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

I'm using retroarch 1.19.1 on my New 3DS and there is a performance degradation when I enable retroachievements, even for gameboy games.
I followed the documentation and removed all cores not being used and that improved loading times but retroachievements are still causing a big impact on performance. Seems to be caching something for the first few minutes and then stabilizes, but if an achievement is going to trigger, there's a lot of stutters.

The 3DS is not a powerful device, but maybe resources are being overused for processing retroachievements.

### Expected behavior

No stutters

### Actual behavior

Lots of stutters

### Steps to reproduce the bug

1. Install RetroArch on 3DS
2. Add games
3. Enable retroachievements 
4. Start a game compatible with retroachievements
5. Massive stutters will be present or game will freeze for several minutes
6. (if retroachievements are disabled, game does not freeze or stutter)

### Bisect Results

N/A

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.19.1

### Environment information

- OS: New Nintendo 3DS
- Compiler: N/A

**Comment by shigutso** [comment]
Indeed it seems to be mostly related to badges being downloaded.
If we can get an option to skip that and keep text-only, would fix the issue.
Or perhaps the download code could be optimized for the 3DS so it won't overwhelm the CPU

However, I found a particular case with a game
If you play Dr. Mario for Game Boy, before starting a run you can select the speed of the game
Every time you change the speed, there's a brief notification from retroachievements, causing mini-stutters
So I don't know if downloading images is the only cause, maybe displaying the images are also having a impact in performance

**Comment by i30817** [comment]
I keep banging on this drum, but everything to do with the notification widget, especially if there is animation going on and the hardware RA is running is not multicore or has threaded tasks disabled is a recipe for 'microstutters'.

This is because of how the widgets display, they have a api to draw something at the start, the end, and a task chosen number in between. It just turns out that to run, these need to interrupt emulation without threaded tasks and I doubt it's very nice for the emulator thread even with them.

This is inevitable but it would be nice to have a alternate take that turns off animations for the more common usecases (sliders, fading in and out etc).
